src/WebSocketHandler.js

```diff
@@ -12,28 +12,37 @@ import WS from './websocket';
 import EventEmitter from 'events';
 
 class WebSocketHandler extends EventEmitter {
+  constructor() {
+    super();
+
+    this.ws = null;
+  }
+
   setup() {
-    this.ws = new WS({ wsURL: helpers.getWSServerURL });
-
-    this.on('is_online', (value) => {
-      if (value) {
-        wallet.onWebsocketOpened();
-        this.ws.emit('reload_data');
-      } else {
-        wallet.onWebsocketBeforeClose();
+    if (this.ws === null) {
```

Comment:
What if it is not null? What do we do?

Reply:
If it's not null we only execute the last line of the method

`return this.ws.setup();`

Run the setup normally

Reply:
I got it. Thanks!